altos-rust/altos-core/src/sched/mod.rs

```diff
@@ -253,12 +254,54 @@ mod tests {
   #[test]
   fn only_critical_task() {
     let _g = test::set_up();
-    let handle_1 = test::create_and_schedule_test_task(512, Priority::Critical, "test task 1");
-    let handle_2 = test::create_and_schedule_test_task(512, Priority::Critical, "test task 2");
+    test::create_and_schedule_test_task(512, Priority::Normal, "test task 1");
+    test::create_and_schedule_test_task(512, Priority::Normal, "test task 2");
     start_scheduler();
-    for i in 0..100 {
-      assert!(test::current_task().unwrap().priority == Priority::Critical);
+    for _i in 0..100 {
+      assert!(test::current_task().unwrap().priority == Priority::Normal);
     }
   }
 
+  #[test]
+  //No task, but scheduler is run
+  fn no_task_sched_run_sched() {
+    let _g = test::set_up();
+    start_scheduler();
+    assert_eq!(test::current_task().unwrap().priority, Priority::__Idle);
+  }
+
+  #[test]
+  //All tasks are blocked
+  fn all_tasks_blocked() {
+    let _g = test::set_up();
+    assert!(test::current_task().is_none());
+    let handle_1 = test::create_and_schedule_test_task(512, Priority::Low, "test task 1");
+    let handle_2 = test::create_and_schedule_test_task(512, Priority::Normal, "test task 2");
+    start_scheduler();
+    assert!(test::current_task().is_some());
+    test::current_task().unwrap().state = State::Blocked;
+    test::current_task().unwrap().delay_type = Delay::Timeout;
```

Comment:
`Delay::Sleep` is probably a better choice here as we're not doing any system ticks, or you can use the `sleep()` system call, though I'm not sure if we'd want to depend on that call in this test